src/Interpreters/Context.cpp

```diff
@@ -696,6 +696,41 @@ void Context::setUser(const String & name, const String & password, const Poco::
     setSettings(*access->getDefaultSettings());
 }
 
+void Context::setUserSecure(const String & name, const Poco::Net::SocketAddress & address)
```

Comment:
In fact this is "insecure" version of the `Context::setUser()` function. I think the name `setUserWithoutCheckingPassword()` would be clearer.

Reply:
The though was that `setUserSecure` assumes that you are calling it from *secure* context, but can be mixed up, agree